src/targets/gpu/mlir.cpp

```diff
@@ -525,6 +530,17 @@ struct mlir_program
         return op;
     }
 
+    void find_target()
+    {
+        std::string tname = get_device_name();
+        // HACK: Since MLIR can't handle the full target name
+        target_name = tname.substr(0, tname.find(':'));
```

Comment:
We can reuse our code from src/targets/gpu/lowering.cpp instead:
```
target_name = trim(split_string(get_device_name(), ':').front());
```

Reply:
done